src/snowflake/connector/connection.py

```diff
@@ -268,8 +268,11 @@ def __init__(self, **kwargs):
 
         self.heartbeat_thread = None
 
-        if "application" not in kwargs and ENV_VAR_PARTNER in os.environ.keys():
-            kwargs["application"] = os.environ[ENV_VAR_PARTNER]
+        if "application" not in kwargs:
+            if ENV_VAR_PARTNER in os.environ.keys():
+                kwargs["application"] = os.environ[ENV_VAR_PARTNER]
+            elif "streamlit" in sys.modules:
```

Comment:
This might me inaccurate for two reasons?
1. "streamlit" won't count in some cases even though a connection is created from a streamlit.
2. "streamlit" is used as long as it's in sys.modules. If a connection is created but it is unrelated to streamlit, it will still be counted.

Is it possible to pass in something when streamlit creates a connection? This is more accurate I think.

Reply:
@sfc-gh-yixie it looks like you were not invited to the meeting yesterday.
Streamlit does not interact with any connectors, users need to manually pipe the data into it and open their own connections. Due to this we couldn't find any other way to do track usage other than what's in this PR.

We all understand that your first comment is a real issue and there's no solution to it unfortunately, but we saw no other way to track Streamlit usage.

Reply:
Thanks for the context.